### PR TITLE
GPUCompiler + LLVM 6 adapt

### PR DESCRIPTION
**Pull request by motabbara**
Not sure if all the active(ctx) are needed.

### CODE REVIEW

**Comment by motabbara** [comment]
Going to fail all of CI because of CUDA incompatability with GPUCompiler.

File: src/Enzyme.jl [hunk 1]
```diff
@@ -178,7 +178,7 @@ Enzyme.autodiff(ReverseWithPrimal, x->x*x, Active(3.0))
     ModifiedBetween = Val(falses_from_args(Val(1), args...))
 
     tt    = Tuple{map(T->eltype(Core.Typeof(T)), args′)...}
-    world = GPUCompiler.codegen_world_age(Core.Typeof(f.val), tt)
+    world = GPUCompiler.tls_world_age() #(Core.Typeof(f.val), tt).min_world
```
Comment (wsmoses) [inline src/Enzyme.jl:181]: I don't think this replacement is equivalent here. I think we need to import the codegen_world_age that was dropped here https://github.com/JuliaGPU/GPUCompiler.jl/commit/3de799afb1e145a72aaa30dc7c29222d32802693

Comment (wsmoses) [reply src/Enzyme.jl:181]: Which, relatedly, can we split off the codegen_world_age refit into its own PR (which presumably should add support for both GPUCompiler 0.19 and 0.20), from the other dependency bumps?

Comment (motabbara) [reply src/Enzyme.jl:181]: So PR1 is GPUCompiler bump and codegen_world_age and PR2 is rest of the stuff?

Comment (wsmoses) [reply src/Enzyme.jl:181]: Yeah, [though the rest may also make sense to split off, depending on how invasive the LLVM stuff is]

**Comment by motabbara** [comment]
Will close and submit two PRs. This doesn't resolve #909 as that needs GPUCompiler 0.21.2 which depends on LLVM 6.

**Pull request closed without merging**